pyaptly/__init__.py

```diff
@@ -1516,10 +1516,17 @@ def is_publish_affected(name, publish):
                 name,
                 publish['distribution']
         ) in state.publishes:
-            for snap in publish['snapshots']:
-                snap_name = snapshot_spec_to_name(cfg, snap)
-                if snap_name in affected_snapshots:
-                    return True
+            try:
+                for snap in publish['snapshots']:
+                    snap_name = snapshot_spec_to_name(cfg, snap)
+                    if snap_name in affected_snapshots:
+                        return True
+            except KeyError:
```

Comment:
Thanks a lot for the PR please consider adding # prgrama: no cover or writing a test for the except case.

```python
except KeyError:  # prgrama: no cover
```

You can run `make test` to see if it is really fixed.